### PR TITLE
Clean up deployment workflow

### PR DESCRIPTION
**Pull request by jake-low**
This PR aims to simplify both development and deploy workflows by making the frontend app easier to run and reducing the differences between what runs in local dev and what gets deployed.

Notable changes:
- `PUBLIC_URL` has been removed; the frontend is now agnostic to where it gets deployed (meaning it uses domain-less URLS like `/static/js/main.12a34bc.js` to fetch required resources). This means you can build one artifact and deploy it anywhere, rather than needing a different artifact for each server you're deploying to (including localhost).
- The Docker image now includes a default `nginx.conf` file. This file is suitable for production use (though it's HTTP, so it should be used along with a TLS-terminating reverse proxy). In a future PR I'll update the [Helm charts in the deploy repo](https://github.com/developmentseed/osmcha-deploy/blob/main/helm/osmcha/values.yaml) to remove the nginx config file that's inlined there, and use this one instead.
- Cleaned up a bunch of old scripts and snippets that were only used when the frontend used to be served from Github Pages.

The changes here go along with some changes to the Django app configuration here: https://github.com/OSMCha/osmcha-django/pull/719

### CODE REVIEW

**Review by willemarcel — APPROVED** [review]
Sorry for the delay, @jake-low! It's working well

**Comment by jake-low** [comment]
No worries, thanks for the review! I'm going to hold off on merging this until OSMCha/osmcha-django#719 is also approved. After that I'll work on the required changes to `osmcha-deploy` as mentioned in that PR.

**Pull request merged**
